### ISSUE TITLE
The integration stop working 

### ISSUE THREAD

**Issue by MiAutomations** [user2684/imou_life#101]
Hello 

After the last Home Assistant update the integration stops working... all the devices/entities are unavailable  

![image](https://github.com/user2684/imou_life/assets/47602733/61d1ace8-a8ae-4ff0-9a78-50013d3bd644)

Can you please check ?
Thank you

**Comment by QNTLO** [comment]
Can you please state for which update this is so I can postpone updating myself. Thanks in advance!

**Comment by davisuco** [comment]
I have same issue after upgrade to 2024.3.0

**Comment by jeffFREEMAN** [comment]
same issue after upgrade

**Comment by talik31** [comment]
Same issue since upgrading to 2024.3.0, reloading the integration isn't solving it.

**Comment by tbkon3** [comment]
I have same issue after upgrade to 2024.3.0

**Comment by ELFHOUSE** [comment]
same for me and the api was ok with ha 12.rc1 but with the update to 12.1 it stoped working

Home Assistant

Core
2024.3.0
Supervisor
2024.03.0
Operating System
12.1
Frontend
20240306.0

**Comment by LeoeLeoeL** [comment]
Same for me

**Comment by sinseman44** [comment]
same for me. 
the cloud API URL (https://openapi.easy4ip.com/openapi) seems to be down ?

**Comment by QNTLO** [comment]
Can anyone who already updated confirm if 2024.3.1 fixes anything?

**Comment by davisuco** [comment]
I just upgraded but still issue is there

**Comment by LeoeLeoeL** [comment]
+1

**Comment by MacPaKKo** [comment]
Same here.
First, I had the problem with the new limits for cloud data adopted from Imou.
Then Imou life stopped working, I suppose after the HA update.

**Comment by user2684** [comment]
Hi all, please always check or share the logs when reporting these errors :-)
Have a look here: https://community.home-assistant.io/t/imou-life-cloud-integration/462439/202
I think this is what you are experiencing. Thanks

**Comment by talik31** [comment]
> Hi all, please always check or share the logs when reporting these errors :-) Have a look here: https://community.home-assistant.io/t/imou-life-cloud-integration/462439/202 I think this is what you are experiencing. Thanks

I don't think it's the same issue, my loss of functionnality appeared right after the upgrade to 2024.3.0 and the imou resource seems to still have 29945 message push & 1.89GB available

![image](https://github.com/user2684/imou_life/assets/49128064/acd69461-d04d-4235-8310-724a84d82192)

I've activated logging and reload the integration and here are the generated logs (where indeed I'm seeing a "APIError: OP1013: Call interface times exceed limit (total)" which is not consistent with imou my resource content) :
```
2024-03-16 12:48:18.723 DEBUG (MainThread) [custom_components.imou_life] Loading entry 84af7115374bbcb2f65a22012715855b
2024-03-16 12:48:18.723 DEBUG (MainThread) [custom_components.imou_life] Setting up device caméra imou entrée (9H0E8F0PBV9C98F)
2024-03-16 12:48:18.724 DEBUG (MainThread) [custom_components.imou_life] Setting API base url to https://openapi.easy4ip.com/openapi
2024-03-16 12:48:18.993 ERROR (MainThread) [imouapi] APIError: OP1013: Call interface times exceed limit (total).
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/site-packages/imouapi/api.py", line 155, in _async_call_api
    await self.async_connect()
  File "/usr/local/lib/python3.12/site-packages/imouapi/api.py", line 117, in async_connect
    data = await self._async_call_api("accessToken", {}, True)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/imouapi/api.py", line 227, in _async_call_api
    raise APIError(error_message)
imouapi.exceptions.APIError: OP1013: Call interface times exceed limit (total).

2024-03-16 12:48:19.071 ERROR (MainThread) [imouapi] APIError: OP1013: Call interface times exceed limit (total).
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/site-packages/imouapi/api.py", line 155, in _async_call_api
    await self.async_connect()
  File "/usr/local/lib/python3.12/site-packages/imouapi/api.py", line 117, in async_connect
    data = await self._async_call_api("accessToken", {}, True)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/imouapi/api.py", line 227, in _async_call_api
    raise APIError(error_message)
imouapi.exceptions.APIError: OP1013: Call interface times exceed limit (total).

2024-03-16 12:48:19.072 ERROR (MainThread) [imouapi] Too many unsuccesful connection attempts
2024-03-16 12:48:19.077 ERROR (MainThread) [custom_components.imou_life] NotConnected: 
Traceback (most recent call last):
  File "/config/custom_components/imou_life/__init__.py", line 89, in async_setup_entry
    await device.async_initialize()
  File "/usr/local/lib/python3.12/site-packages/imouapi/device.py", line 182, in async_initialize
    device_array = await self._api_client.async_api_deviceBaseDetailList([self._device_id])
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/imouapi/api.py", line 279, in async_api_deviceBaseDetailList
    return await self._async_call_api(api, payload)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/imouapi/api.py", line 160, in _async_call_api
    raise NotConnected()
imouapi.exceptions.NotConnected

2024-03-16 12:48:19.078 ERROR (MainThread) [homeassistant.config_entries] Error setting up entry caméra imou entrée for imou_life
Traceback (most recent call last):
  File "/config/custom_components/imou_life/__init__.py", line 89, in async_setup_entry
    await device.async_initialize()
  File "/usr/local/lib/python3.12/site-packages/imouapi/device.py", line 182, in async_initialize
    device_array = await self._api_client.async_api_deviceBaseDetailList([self._device_id])
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/imouapi/api.py", line 279, in async_api_deviceBaseDetailList
    return await self._async_call_api(api, payload)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/imouapi/api.py", line 160, in _async_call_api
    raise NotConnected()
imouapi.exceptions.NotConnected

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 504, in async_setup
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/imou_life/__init__.py", line 92, in async_setup_entry
    raise ImouException() from exception
imouapi.exceptions.ImouException
```

For additional info here are my versions:
Core 2024.3.0
Supervisor 2024.03.0
Operating System 12.1
Interface utilisateur 20240306.0
imou life version 1.0.15 installed via HACS

**Comment by umarino** [comment]
I have the same issue but for me it depends on IMOU URL API modification probably for the management of the limit repricing.

Take a look to the API URL:
https://open.imoulife.com/book/http/develop.html

**Comment by talik31** [comment]
Issue is "kinda solved" for me.
1) I've configured the cameras with onvif integration. There's an issue with onvif integration with my imou cam model where all the cameras are identifying with the same false mac adress -> solved in [this comment](https://github.com/home-assistant/core/issues/97258#issuecomment-1728553364)
2) I've deactivated the camera video stream entity from IMOU integration.
3) Changed the api-url from "https://openapi.easy4ip.com/openapi" to "https://openapi-fk.easy4ip.com:443/openapi" in .storage\core.config_entries & reboot HA
4) Put 1USD in my account balance to keep using IMOU Integration for the detection activation then reload the integration and it's working fine again

So I guess the issue is on IMOU side & that setting a minimum amount of money in the account balance unstuck the situation. I haven't tried to get rid of the IMOU dev account & reset it from scratch but maybe this option is also working.

**Comment by peppeman75** [comment]
thank, working great!!

**Comment by peppeman75** [comment]
I solved it like this:
taken from the Imou app the name of the camera
found the ip inside the router
added the camera with ONVIF integration
User: admin
Password: from the Imou Android program read "device label" --> security code
once this is done the camera works in Onvif.
HI
Joseph

**Comment by umarino** [comment]
> Issue is "kinda solved" for me.
> 
> 1. I've configured the cameras with onvif integration. There's an issue with onvif integration with my imou cam model where all the cameras are identifying with the same false mac adress -> solved in [this comment](https://github.com/home-assistant/core/issues/97258#issuecomment-1728553364)
> 2. I've deactivated the camera video stream entity from IMOU integration.
> 3. Changed the api-url from "https://openapi.easy4ip.com/openapi" to "https://openapi-fk.easy4ip.com:443/openapi" in .storage\core.config_entries & reboot HA
> 4. Put 1USD in my account balance to keep using IMOU Integration for the detection activation then reload the integration and it's working fine again
> 
> So I guess the issue is on IMOU side & that setting a minimum amount of money in the account balance unstuck the situation. I haven't tried to get rid of the IMOU dev account & reset it from scratch but maybe this option is also working.

Great bro, solved for me also just charging 1$ on IMOU Dev!
Thks a lot :)

**Comment by user2684** [comment]
Thanks for the comments guys. Just to confirm, has the URL change fixed the problem or charging with $ the account or both? Thanks

**Comment by davisuco** [comment]
I have the same question. What would be the minimum change needed to make it work again?

Thanks!

**Comment by umarino** [comment]
> Thanks for the comments guys. Just to confirm, has the URL change fixed the problem or charging with $ the account or both? Thanks

both :)

**Comment by fcaronte** [comment]
> Issue is "kinda solved" for me.
> 1) I've configured the cameras with onvif integration. There's an issue with onvif integration with my imou cam model where all the cameras are identifying with the same false mac adress -> solved in [this comment](https://github.com/home-assistant/core/issues/97258#issuecomment-1728553364)
> 2) I've deactivated the camera video stream entity from IMOU integration.
> 3) Changed the api-url from "https://openapi.easy4ip.com/openapi" to "https://openapi-fk.easy4ip.com:443/openapi" in .storage\core.config_entries & reboot HA
> 4) Put 1USD in my account balance to keep using IMOU Integration for the detection activation then reload the integration and it's working fine again
> 
> So I guess the issue is on IMOU side & that setting a minimum amount of money in the account balance unstuck the situation. I haven't tried to get rid of the IMOU dev account & reset it from scratch but maybe this option is also working.

I add 1dollar and try putting new link on the login page but is not work for me, but can't find the file in your location

**Comment by tbkon3** [comment]
I fix as 

> Issue is "kinda solved" for me.
> 
> 1. I've configured the cameras with onvif integration. There's an issue with onvif integration with my imou cam model where all the cameras are identifying with the same false mac adress -> solved in [this comment](https://github.com/home-assistant/core/issues/97258#issuecomment-1728553364)
> 2. I've deactivated the camera video stream entity from IMOU integration.
> 3. Changed the api-url from "https://openapi.easy4ip.com/openapi" to "https://openapi-fk.easy4ip.com:443/openapi" in .storage\core.config_entries & reboot HA
> 4. Put 1USD in my account balance to keep using IMOU Integration for the detection activation then reload the integration and it's working fine again
> 
> So I guess the issue is on IMOU side & that setting a minimum amount of money in the account balance unstuck the situation. I haven't tried to get rid of the IMOU dev account & reset it from scratch but maybe this option is also working.

Fixed for me too with:
1) Put 1USD in my account balance to keep using IMOU Integration
2) Changed the api-url from "https://openapi.easy4ip.com/openapi" to "[https://openapi-fk.easy4ip.com:443/openapi](https://openapi-fk.easy4ip.com/openapi)" in .storage\core.config_entries & reboot HA

**Comment by Flizs** [comment]
> > Le problème est "un peu résolu" pour moi.
> > 
> > 1. J'ai configuré les caméras avec l'intégration ONVIF. Il y a un problème avec l'intégration d'Onvif avec mon modèle de caméra imou où toutes les caméras s'identifient avec la même fausse adresse Mac -> résolu dans [ce commentaire](https://github.com/home-assistant/core/issues/97258#issuecomment-1728553364)
> > 2. J'ai désactivé l'entité de flux vidéo de la caméra de l'intégration IMOU.
> > 3. Modification de l'URL de l'API de " https://openapi.easy4ip.com/openapi " à " https://openapi-fk.easy4ip.com:443/openapi " dans .storage\core.config_entries et redémarrage HA
> > 4. Mettez 1 USD sur le solde de mon compte pour continuer à utiliser l'intégration IMOU pour l'activation de la détection, puis rechargez l'intégration et cela fonctionne à nouveau correctement.
> > 
> > Je suppose donc que le problème vient du côté d'IMOU et que la définition d'un montant minimum d'argent dans le solde du compte a permis de débloquer la situation. Je n'ai pas essayé de me débarrasser du compte de développement IMOU et de le réinitialiser à partir de zéro, mais peut-être que cette option fonctionne également.
> 
> Corrigé pour moi aussi avec :
> 
> 1. Mettez 1 USD sur le solde de mon compte pour continuer à utiliser l'intégration IMOU
> 2. Modification de l'URL de l'API de " https://openapi.easy4ip.com/openapi " à " [https://openapi-fk.easy4ip.com:443/openapi](https://openapi-fk.easy4ip.com/openapi) " dans .storage\core.config_entries et redémarrage HA

Je vous confirme qu'en suivant cette méthode cela refonctionne bien.

**Comment by hernnyy** [comment]
> > > Le problème est "un peu résolu" pour moi.
> > > 
> > > 1. J'ai configuré les caméras avec l'intégration ONVIF. Il y a un problème avec l'intégration d'Onvif avec mon modèle de caméra imou où toutes les caméras s'identifient avec la même fausse adresse Mac -> résolu dans [ce commentaire](https://github.com/home-assistant/core/issues/97258#issuecomment-1728553364)
> > > 2. J'ai désactivé l'entité de flux vidéo de la caméra de l'intégration IMOU.
> > > 3. Modification de l'URL de l'API de " https://openapi.easy4ip.com/openapi " à " https://openapi-fk.easy4ip.com:443/openapi " dans .storage\core.config_entries et redémarrage HA
> > > 4. Mettez 1 USD sur le solde de mon compte pour continuer à utiliser l'intégration IMOU pour l'activation de la détection, puis rechargez l'intégration et cela fonctionne à nouveau correctement.
> > > 
> > > Je suppose donc que le problème vient du côté d'IMOU et que la définition d'un montant minimum d'argent dans le solde du compte a permis de débloquer la situation. Je n'ai pas essayé de me débarrasser du compte de développement IMOU et de le réinitialiser à partir de zéro, mais peut-être que cette option fonctionne également.
> > 
> > 
> > Corrigé pour moi aussi avec :
> > 
> > 1. Mettez 1 USD sur le solde de mon compte pour continuer à utiliser l'intégration IMOU
> > 2. Modification de l'URL de l'API de " https://openapi.easy4ip.com/openapi " à " [https://openapi-fk.easy4ip.com:443/openapi](https://openapi-fk.easy4ip.com/openapi) " dans .storage\core.config_entries et redémarrage HA
> 
> Je vous confirme qu'en suivant cette méthode cela refonctionne bien.

It work for me just charging 1usd and restarting the HA

**Comment by umarino** [comment]
![image](https://github.com/user2684/imou_life/assets/61564645/2ad47280-76a9-4e9c-b86b-d95d0b89dc0d)

**Comment by Matuuu96** [comment]
There is no need to change API url, just put 1 USD to your balance and restart HA.

**Comment by VangelisP** [comment]
> There is no need to change API url, just put 1 USD to your balance and restart HA.

I can confirm that somehow, magically, this works.  
I have 2 cameras that used to work with the version older than 2024.3.0  
Thanks!

**Comment by fcaronte** [comment]
Umh i can't understand why all this requests, seem as I can't use for free anymore on home assistant I have the streaming trough ONVIF and I use only the enable\disable motion movement and close\open camera on two camera around 10 request at day or lower and can't understand why say over 1000... Now on imou integration I disabled all other functions and keep only the two I use let's see if is better ![Screenshot_20240324_101151_Chrome.jpg](https://github.com/user2684/imou_life/assets/26366403/18b30e76-9ee1-482e-9b19-d23914e90c91)

![Screenshot_20240324_101132_Chrome.jpg](https://github.com/user2684/imou_life/assets/26366403/2dd9b872-8d89-48b0-8dc7-e6ab4ae51622)

**Comment by user2684** [comment]
@fcaronte I have a similar use case, I use the integration only for programmatically enable/disable motion detection when the main alarm is set. Have you already tried increasing the polling interval? I mean, even to 6 hours if you don't need to refresh the data. Also please find all the details regarding the calls the integration makes here: https://community.home-assistant.io/t/imou-life-cloud-integration/462439/202

**Comment by fcaronte** [comment]
> @fcaronte I have a similar use case, I use the integration only for programmatically enable/disable motion detection when the main alarm is set. Have you already tried increasing the polling interval? I mean, even to 6 hours if you don't need to refresh the data. Also please find all the details regarding the calls the integration makes here: https://community.home-assistant.io/t/imou-life-cloud-integration/462439/202

Nope, I need to give  a check, thanks

**Comment by kaesar3arg** [comment]
Hello, I was able to make it work by going to https://open.imoulife.com/consoleNew/myApp/appInfo, edit my app information, select Account Info H5/Web to modify something, save the changes. I went to IMOU life eb HA and recharged and walked away. Greetings

**Comment by foekpham** [comment]
> Hello, I was able to make it work by going to https://open.imoulife.com/consoleNew/myApp/appInfo, edit my app information, select Account Info H5/Web to modify something, save the changes. I went to IMOU life eb HA and recharged and walked away. Greetings

This worked for me 👍

**Comment by woody4165** [comment]
> Hello, I was able to make it work by going to https://open.imoulife.com/consoleNew/myApp/appInfo, edit my app information, select Account Info H5/Web to modify something, save the changes. I went to IMOU life eb HA and recharged and walked away. Greetings

Thanks for the info.
You mean to select only H5/web or can I leave also PC Client and Android App ?

**Comment by kaesar3arg** [comment]
Hello, just add the item, h5/web to enable the save button. I left the rest
as they were. greetings

**Comment by gRosmorduc** [comment]
> Hello, I was able to make it work by going to https://open.imoulife.com/consoleNew/myApp/appInfo, edit my app information, select Account Info H5/Web to modify something, save the changes. I went to IMOU life eb HA and recharged and walked away. Greetings

Works for me too !! Thanks a lot !

**Comment by wingmanx** [comment]
> Hello, I was able to make it work by going to https://open.imoulife.com/consoleNew/myApp/appInfo, edit my app information, select Account Info H5/Web to modify something, save the changes. I went to IMOU life eb HA and recharged and walked away. Greetings

Thanks! Worked for me.

**Comment by alexs197912** [comment]
> Hello, I was able to make it work by going to https://open.imoulife.com/consoleNew/myApp/appInfo, edit my app information, select Account Info H5/Web to modify something, save the changes. I went to IMOU life eb HA and recharged and walked away. Greetings

Flagging H5/web work also for me.

**Comment by user2684** [comment]
In my experience changing the app info is not necessary. I just charged the account, reloaded the integration in HA (not even restarted it) and started working back again. It *might* be that changing the app info and generating a new appId would make the counter starting over but still not required from what I can see. Am I the only one?

**Comment by mabusdogma** [comment]
Could it be the same issue?

`. Registrador: homeassistant.config_entries
Fuente: config_entries.py:551
Ocurrió por primera vez: 1:37:30 a. m. (9 ocurrencias)
Último inicio de sesión: 7:13:56 p. m.

Error setting up entry Camara de salon for imou_life
Traceback (most recent call last):
  File "/config/custom_components/imou_life/__init__.py", line 89, in async_setup_entry
    await device.async_initialize()
  File "/usr/local/lib/python3.12/site-packages/imouapi/device.py", line 182, in async_initialize
    device_array = await self._api_client.async_api_deviceBaseDetailList([self._device_id])
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/imouapi/api.py", line 279, in async_api_deviceBaseDetailList
    return await self._async_call_api(api, payload)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/imouapi/api.py", line 160, in _async_call_api
    raise NotConnected()
imouapi.exceptions.NotConnected

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 551, in async_setup
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/imou_life/__init__.py", line 92, in async_setup_entry
    raise ImouException() from exception
imouapi.exceptions.ImouException   `

**Comment by ioman2** [comment]
It is simple to solve the problem with "API error":
If you are from the area:
East Asia Data Center https://openapi-sg.easy4ip.com:443/openapi
Central Europe Data Center https://openapi-fk.easy4ip.com:443/openapi
Western America Data Center https://openapi-or.easy4ip.com:443/openapi

My information is from "https://open.imoulife.com/book/http/develop.html"
I'm from Central Europe, so I put https://openapi-fk.easy4ip.com:443/openapi and it worked perfectly.

**Comment by Andreasav** [comment]
> Hello, I was able to make it work by going to https://open.imoulife.com/consoleNew/myApp/appInfo, edit my app information, select Account Info H5/Web to modify something, save the changes. I went to IMOU life eb HA and recharged and walked away. Greetings

GREAT it works perfect
